src/uta/parsers/seqrecord.py

```diff
@@ -20,32 +19,38 @@ def features_by_type(self) -> dict[str, list]:
         result = defaultdict(list)
         for feat in self._sr.features:
             result[feat.type].append(feat)
-        self.validate_features_by_type(result)
         return result
 
-    @staticmethod
-    def validate_features_by_type(features: dict[str, list]) -> None:
-        """Raise exceptions if feature mapping is invalid."""
-        if "CDS" in features and len(features["CDS"]) > 1:
-            raise SeqRecordFeatureError("Expected one `CDS` feature at most")
-        if "gene" not in features or len(features["gene"]) != 1:
-            raise SeqRecordFeatureError("Expected exactly one `gene` feature")
-
     @cached_property
     def cds_feature(self) -> Optional[SeqFeature]:
         """
         Returns the CDS feature for any coding transcript, None for any non-coding transcript.
+        Some NCBI records will contain multiple CDS features. In these one CDS describes a protein
+        with accession and protein sequence, they other CDS features describes a pseudogene. This method
+        will preferentially choose the CDS feature with a protein sequence.
         """
         cds_features = self.features_by_type.get("CDS")
         if cds_features is None:
             return None
         else:
-            return cds_features[0]
+            # Prefer CDS with protein accession and translated sequence.
+            translated_cds_features = [
```

Comment:
Older genbank files can have multiple CDS. This method filters out the CDS feature that does not have information about the protein.